src/bootstrap/tool.rs

```diff
@@ -696,7 +548,6 @@ impl Step for Cargo {
                 tool: name,
                 mode: Mode::ToolRustc,
                 path,
-                is_optional_tool: true,
```

Comment:
This is intentionally set to be optional. These credential programs had some risk that they wouldn't compile in all environments. I suppose if you can get CI to pass on all targets, then it should be ok. But I want to point out that this is a risky change, and I'm a little nervous about doing this.